### PR TITLE
Add Basics with Bioawk blog post -- Sahil Verma

### PR DESCRIPTION
**Pull request by sdverma89**
A blog post including a small/quick intro to Awk, segueing into the usefulness of Bioawk as a bioinformatician with biological file formats.

### CODE REVIEW

File: _posts/2020-11-09-Basics-With-Bioawk.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title:  "Basics with Bioawk!"
+date:   2020-11-09 4:45:00 -0800
+categories: Bioawk
```
Comment (naupaka) [inline _posts/2020-11-09-Basics-With-Bioawk.md:5]: You could also add awk and perhaps bash as category tags.

File: _posts/2020-11-09-Basics-With-Bioawk.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+Bioawk also has some built in functionality that bioinformaticians in particular will love so much, they'll come back for seconds! One of these functions is to reverse compliment a sequence using `revcomp()`.
+
+`$ bioawk -c fastx '{print revcomp($seq)}' example.fasta | head -n 10`
+
+These are but a few of the ways you can introduce the Bioawk cuisine to your daily ~meals~ workflow.
+
+## Final Tasting Notes
+
+Sometimes you head to your ~kitchen~ desk and you see a lot of things in your ~fridge~ code that are not appealing, consider ~ordering takeout~ googling stack overflow, and go back to the ~fridge~ code knowing very well that nothing has changed. But sometimes you _do_ check stack overflow and you realize there's a comment detailing Bioawk. Do yourself a favor and, consider adding Bioawk to your workflow. As soon as you start picking up the basics of Awk, you'll find yourself generating bioinformatics reports with ease using Bioawk!
```
Comment (naupaka) [inline _posts/2020-11-09-Basics-With-Bioawk.md:62]: Very funny!

**Review by naupaka — DISMISSED** [review]
Great job overall, I just had a few suggestions for improvements.

File: _posts/2020-11-09-Basics-With-Bioawk.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+Awk programs are, generally, built using the following recipe:
+
+`pattern {action}`
+
+The _pattern_ is an expression or regular expression patter, allowing people to filter for matched expressions, so that those only those patterns are acted upon. These _pattern-actions_ can be linked together into a chain by semicolons. By combining these two different concepts, _records_ and _fields_ with _pattern-actions_, Awk is able survey a file line by line and split those lines into fields, compare those lines and fields to a pattern, and perform actions on those patters to manipulate data and generate reports.
```
Comment (sbdarrow) [inline _posts/2020-11-09-Basics-With-Bioawk.md]: It looks like there might be a slight typo here. You type "patter" instead of "pattern".

Comment (sbdarrow) [inline _posts/2020-11-09-Basics-With-Bioawk.md]: I think maybe you forgot a word here as well. You write "Awk is able survey ...", do you mean to put "Awk is able to survey ..." ?

File: _posts/2020-11-09-Basics-With-Bioawk.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+Some common Awk functions are to filter reports and fields using RegEx, splitting strings (s) into tabular "chunks" with a specificed delimiter (d) and placing them into an array (x) via `split (s, x, d)`, `print` the content of the tabular data, changing the case of strings in a field via `tolower(s)` or `toupper(s)`
+
+Now that we've lightly sautéed the awk, so to speak, let's add the "bio" back to "Bioawk."
+
+## Bioawk: Making Biological Formats More Awk Since 2015
```
Comment (sbdarrow) [inline _posts/2020-11-09-Basics-With-Bioawk.md:25]: Fantastic header

File: _posts/2020-11-09-Basics-With-Bioawk.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    1:seqname 2:source 3:feature 4:start 5:end 6:score 7:filter 8:strand
+    9:group 10:attribute
+
+fastx:
+    1:name 2:seq 3:qual 4:comment
+```
+Bioawk code ~tastes~ is built similarly to Awk, but by using bioawk's built in tools, we can specify the filetype we are using with the following syntax, using FASTX as an example: `bioawk -c fastx`
+
+We will continue to FASTA/FASTQ files as an example moving forward. As mentioned above, Bioawk has built in tooling to scan a file and "chunk" parts of the sequence file into their appropriate fields, the name of the sequence (`$name`), the sequence itself `($seq)`, comments about the sequence such as what the gene does, what chromosome it is on, etc. (`$comment`), and finally FASTQ files have an additional field for quality data about the sequence (`qual`). If we wanted to print the comments of all the sequences in a FASTA file, we could write the following:
```
Comment (sbdarrow) [inline _posts/2020-11-09-Basics-With-Bioawk.md]: I think here you might have omitted a word as well. You write "we will continue to FAST/FASTQ files ...", do you mean to write "we will continue to use FAST/FASTQ files ..." ?

File: _posts/2020-11-09-Basics-With-Bioawk.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+`$ bioawk -c fasta '{print $comment}' example.fasta`
+
+We can use Bioaw's built-in Awk functions to further split the comments into tabular data and print them with the following recipe:
```
Comment (sbdarrow) [inline _posts/2020-11-09-Basics-With-Bioawk.md]: Small typo: "Bioaw's"

**Review by sbdarrow — CHANGES_REQUESTED** [review]
Hi Sahil, 
This was overall a very good blog post and it was very helpful! Reading through it reminded me of the basics of Bioawk, so a job well done to you. There were just a few things I noticed that need fixing and they all were dealing with grammar. I think the content of your post was great and everything was organized nicely. Once the grammar mistakes are fixed, the blog post should be all good to go!

**Review by naupaka — CHANGES_REQUESTED** [review]
Nice job! Once you have addressed all of @sbdarrow's suggestions, I think this is ready to merge.

**Comment by naupaka** [comment]
@sdverma89 please make the suggested changes so that we can merge this and get you the point

File: _posts/2020-11-09-Basics-With-Bioawk.md (hunk 3 above)
Comment (sdverma89) [reply _posts/2020-11-09-Basics-With-Bioawk.md]: Fixed in: 053919abeef358d4e41fc393829bdd58bbe8f8e9

**Comment by sdverma89** [comment]
@naupaka suggestions above have been added here: 053919abeef358d4e41fc393829bdd58bbe8f8e9

**Review by naupaka — APPROVED** [review]

**Comment by naupaka** [comment]
Looks good 👍

**Comment by naupaka** [comment]
@sdverma89 Check for rendering issues: https://2020f-usfca-btec-640.github.io/bioawk/2020/11/09/Basics-With-Bioawk.html

Submit a new PR with a fix -- it looks like the pipe character is confusing markdown. You should fix that. Submit a new PR and I'll merge it right away.

File: _posts/2020-11-09-Basics-With-Bioawk.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+We will continue to use FASTA/FASTQ files as an example moving forward. As mentioned above, Bioawk has built in tooling to scan a file and "chunk" parts of the sequence file into their appropriate fields, the name of the sequence (`$name`), the sequence itself `($seq)`, comments about the sequence such as what the gene does, what chromosome it is on, etc. (`$comment`), and finally FASTQ files have an additional field for quality data about the sequence (`qual`). If we wanted to print the comments of all the sequences in a FASTA file, we could write the following:
+
+`$ bioawk -c fasta '{print $comment}' example.fasta`
+
+We can use Bioawk's built-in Awk functions to further split the comments into tabular data and print them with the following recipe:
+
+`$ bioawk -c fasta '{split($comment,a,"|"); print a}' example.fasta | head -n 10`
+
+The above splits the comment of the sequences into tabular chunks, using "|" as the delimiter to separate chunks, and places them into an array, "a" and then print the first 10 rows of data using `head`.
```
Comment (naupaka) [inline _posts/2020-11-09-Basics-With-Bioawk.md:52]: Markdown thinks this line is a table...

File: _posts/2020-11-09-Basics-With-Bioawk.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title:  "Basics with Bioawk!"
+date:   2020-11-09 4:45:00 -0800
+categories: Bioawk
+---
+
+# Basics with ~Babish~ Bioawk!
```
Comment (naupaka) [inline _posts/2020-11-09-Basics-With-Bioawk.md:8]: I think you may need two tildes on each side to have it render as a strikethrough
https://www.markdownguide.org/tools/jekyll/

**Pull request merged**